_posts/2024-09-29-security-releases.md

```diff
@@ -0,0 +1,134 @@
+
+Passing untrusted user input&mdash;even after sanitizing it&mdash;to `SendStream.redirect()` can execute untrusted code.
+
+**Affected versions**: `< 0.19.0`
+
+**Patched versions**: `>=0.19.0`
+
+This vulnerability was discovered during the [OSTIF audit of Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express security triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+For more details, see [GHSA-m6fv-jmcg-4jfg](https://github.com/pillarjs/send/security/advisories/GHSA-m6fv-jmcg-4jfg).
+
+
+## Moderate severity vulnerability CVE-2024-43800 in serve-static middleware
+
+The **[serve-static](https://www.npmjs.com/package/serve-static) middleware module is vulnerable to template injection that can lead to vulnerability to cross-site scripting (XSS) attack**.
+
+Passing untrusted user input&mdash;even after sanitizing it&mdash;to `redirect()` can execute untrusted code.
+
+**Affected versions**: 
+- `< 1.16.0`
+- `>=2.0.0`, `<2.1.0`
+
+**Patched versions**:
+- `>=1.16.0`
+- `>=2.1.0`
+
+This vulnerability was discovered during the [OSTIF audit of Express](https://github.com/expressjs/security-wg/issues/6) and was mitigated by [the Express security triage team](https://github.com/expressjs/security-wg?tab=readme-ov-file#security-triage-team).
+
+For more details, see [GHSA-cm22-4g7w-348p](https://github.com/expressjs/serve-static/security/advisories/GHSA-cm22-4g7w-348p)
+
+
+## Moderate severity vulnerability CVE-2024-45296 in path-to-regexp utility module
+
+The **[path-to-regexp](https://www.npmjs.com/package/path-to-regexp) utility module is vulnerable to regular expression denial of service (ReDoS) attack**.
+
+A bad regular expression is generated any time you have two parameters within a single segment, separated by something that is not a period (`.`). For example, `/:a-:b`.
+
+Using `/:a-:b` will produce the regular expression `/^\/([^\/]+?)-([^\/]+?)\/?$/`. This can be exploited by a path such as `/a${'-a'.repeat(8_000)}/a`. [OWASP](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS) has a good example of why this occurs, but in essence, the `/a` at the end ensures this route would never match, but due to naive backtracking it will still attempt every combination of the `:a-:b` on the repeated 8,000 `-a`.
+
+Because JavaScript is single-threaded and regex matching runs on the main thread, poor performance will block the event loop and can lead to a DoS. In local benchmarks, exploiting the unsafe regex will result in performance that is over 1000x worse than the safe regex. In a more realistic environment, using Express v4 and ten concurrent connections results in an average latency of ~600ms vs 1ms.
+
+**Affected versions**:
+- `>=4.0.0`, `<8.0.0`
+- `>=0.2.0`, `<1.9.0`
+- `<0.1.10`
+- `>=2.0.0`, `<3.3.0`
+
+
+**Patched versions**:
+- `>8.0.0`
+- `>1.9.0`
+- `>=0.1.10`
+- `>3.3.0`
```

Comment:
Shouldn't it be:
```
- `>=8.0.0`
- `>=1.9.0`
- `>=0.1.10`
- `>=3.3.0`
```

Reply:
Updated in [bcd9dd3](https://github.com/expressjs/expressjs.com/pull/1607/commits/bcd9dd31f58f927365fa053a5050f6fcf1bd5a68), based on https://github.com/pillarjs/path-to-regexp/security/advisories/GHSA-9wv6-86v2-598j